fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -186,22 +199,54 @@ void analyzeHighOrderFunction(Visitor visitor, Expr expression, Scope scope) {
+                Expr expr = expression.getChild(i);
+                bottomUpAnalyze(visitor, expr, scope);
+                if (expr instanceof NullLiteral) {
+                    expr.setType(Type.ARRAY_INT); // Let it have item type.
+                }
+                if (!expr.getType().isArrayType()) {
+                    throw new SemanticException(i + "th lambda input should be arrays.");
+                }
+                Type itemType = ((ArrayType) expr.getType()).getItemType();
+                if (itemType == Type.NULL) { // Since slot_ref with Type.NULL is rewritten to Literal in toThrift(),
+                    // rather than a common columnRef, so change its type here.
+                    itemType = Type.BOOLEAN;
+                }
+                scope.putLambdaInput(new PlaceHolderExpr(-1, expr.isNullable(), itemType));
+            }
+        } else {
+            // map_apply(func, map)
+            if (!(expression.getChild(0).getChild(0) instanceof MapExpr)) {
+                throw new SemanticException("The right part of map lambda function (" +
+                        expression.getChild(0).toSql() + ") should have key and value arguments");
+            }
+            if (expression.getChild(0).getChildren().size() != 3) {
+                throw new SemanticException("The left part of map lambda function (" +
+                        expression.getChild(0).toSql() + ") should have 2 arguments, but there are "
+                        + (expression.getChild(0).getChildren().size() - 1) + " arguments");
+            }
+            Expr expr = expression.getChild(1);
             bottomUpAnalyze(visitor, expr, scope);
             if (expr instanceof NullLiteral) {
-                expr.setType(Type.ARRAY_INT); // Let it have item type.
+                expr.setType(Type.ANY_MAP); // Let it have item type.
             }
-            if (!expr.getType().isArrayType()) {
-                throw new SemanticException(i + "th lambda input should be arrays.");
+            if (!expr.getType().isMapType()) {
+                throw new SemanticException("Lambda inputs should be maps.");
             }
-            Type itemType = ((ArrayType) expr.getType()).getItemType();
-            if (itemType == Type.NULL) { // Since slot_ref with Type.NULL is rewritten to Literal in toThrift(),
+            Type keyType = ((MapType) expr.getType()).getKeyType();
+            Type valueType = ((MapType) expr.getType()).getValueType();
+            if (keyType == Type.NULL) { // Since slot_ref with Type.NULL is rewritten to Literal in toThrift(),
                 // rather than a common columnRef, so change its type here.
-                itemType = Type.BOOLEAN;
+                keyType = Type.BOOLEAN;
             }
-            scope.putLambdaInput(new PlaceHolderExpr(-1, expr.isNullable(), itemType));
+            if (valueType == Type.NULL) {
+                valueType = Type.BOOLEAN;
```

Comment:
assign it Type.NULL rather than Type.BOOLEAN, and hack them in `ScalarOperatorToExpr`

Reply:
ok, deleted.